consumer/consumererror/permanent.go

```diff
@@ -14,27 +14,60 @@
 
 package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
 
-import "errors"
+import (
+	"errors"
+	"strconv"
+)
 
-// permanent is an error that will be always returned if its source
+// NO_DATA_COUNT indicates an unknown quantity of telemetry items.
+const NO_DATA_COUNT = -1
+
+// Permanent is an error that will always be returned if its source
 // receives the same inputs.
-type permanent struct {
-	err error
+type Permanent struct {
+	error
+	successful int
+	failed     int
```

Comment:
`accepted`/`rejected` could also be good terminology here, though to me those terms feel like they apply more to ingesting the data into a backend, where `successful`/`failed` I think apply more broadly to failures anywhere in the pipeline.

Reply:
Why do we need both successful and failed? Caller always know what's the total number of items. I think we can follow what's defined in [the proto](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/collector/logs/v1/logs_service.proto#L69) and keep just one `rejected`

Reply:
You're right. I took a look at where this is used, and having just the `rejected` field is sufficient. I've update the fields accordingly.